### DISCUSSION TITLE
Add a custom Statsd Client on AWS Provider to push metrics directly to Cloudwatch Metrics.

### DISCUSSION THREAD

**Discussion by mariotaddeucci** [apache/airflow#23346] [Ideas] [answered]
Airflow supports to use a custom StatsdClient setting the config `statsd_custom_client_path` on `metrics` section.
By extending `statsd.StatsClient` we can override the `constructor` and the `_send` method, to send directly to cloudwatch metrics, simplifying the way to export metrics to AWS.
This class can be added on `apache-airflow-providers-amazon` adding the `metrics` submodule.

```
[metrics]
statsd_custom_client_path = airflow.providers.amazon.aws.metrics.cloudwatch.StatsClient
```

If t makes sense, I am willing to submit a PR!

**Comment by potiuk** [comment, marked as answer]
I think it would be much more powerful to comment and vote on the https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-49+OpenTelemetry+Support+for+Apache+Airflow - this will be way more powerful and once it is implemented the old way of collecting metrics will be deprecated.

CloudWatch has full support for OpenTelemetry https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-open-telemetry.html

The "metrics" replacement with OpenTelemetry in Airflow shoudl be easy to implement for 2.4.0 once discsusion is complete and voting done and this is the first time any modification like this would happen in 2.4.0 only so I think it makes much more sense.
